File: Easy/415. Add Strings/addStrings.cpp
class Solution {
public:
    string addStrings(string num1, string num2) {
        string ans = "";
        int len1 = num1.size();
        int len2 = num2.size();
        int i = len1 - 1;
        int j = len2 - 1;
        int carry = 0;
        while(i >= 0 || j >= 0){
            int a = i >= 0? num1[i--] - '0' : 0;
            int b = j >= 0? num2[j--] - '0' : 0;
            int sum = a + b + carry;
            ans.insert(ans.begin(), sum % 10 + '0');
            carry = sum / 10;
        }
        return carry == 0? ans : "1" + ans;
    }
};

File: Easy/415. Add Strings/addStrings.java
class Solution {
    public String addStrings(String num1, String num2) {
        int n1 = num1.length();
        int n2 = num2.length();
        int i = n1 - 1;
        int j = n2 - 1;
        int carry = 0;
        StringBuilder ans = new StringBuilder();
        while (i >= 0 || j >= 0) {
            int a = i >= 0? num1.charAt(i--) - '0': 0;
            int b = j >= 0? num2.charAt(j--) - '0': 0;
            int sum = a + b + carry;
            ans.append(sum % 10);
            carry = sum / 10;
        }
        if (carry > 0) {
            ans.append(1);
        }
        return ans.reverse().toString();
    }
}